src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -4883,8 +4879,210 @@ internal static Type GetTypeFromCLSIDImpl(Guid clsid, String server, bool throwO
 
         #endregion
 
-        #region COM
-        #endregion
+#if FEATURE_COMINTEROP
+        [System.Security.SecuritySafeCritical]
+        private Object ForwardCallToInvokeMember(
+            String memberName,
+            BindingFlags flags,
+            Object target,
+            Object[] aArgs, // in/out - only byref values are in a valid state upon return
+            int[] aArgsIsByRef,
+            int[] aArgsWrapperTypes, // _maybe_null_
+            Type[] aArgsTypes,
+            Type retType)
+        {
+            Debug.Assert(
+                aArgs.Length == aArgsIsByRef.Length
+                && aArgs.Length == aArgsTypes.Length
+                && (aArgsWrapperTypes == null || aArgs.Length == aArgsWrapperTypes.Length), "Input arrays should all be of the same length");
+
+            int cArgs = aArgs.Length;
+
+            // Handle arguments that are passed as ByRef and those
+            // arguments that need to be wrapped.
+            ParameterModifier[] aParamMod = null;
+            if (cArgs > 0)
+            {
+                ParameterModifier paramMod = new ParameterModifier(cArgs);
+                for (int i = 0; i < cArgs; i++)
+                {
+                    Debug.Assert(aArgsIsByRef[i] == 0 || aArgsIsByRef[i] == 1, "ByRef array contains invalid value");
+                    paramMod[i] = aArgsIsByRef[i] == 1;
+                }
+
+                aParamMod = new ParameterModifier[] { paramMod };
+                if (aArgsWrapperTypes != null)
+                {
+                    WrapArgsForInvokeCall(aArgs, aArgsWrapperTypes);
+                }
+            }
+
+            Object ret = null;
+            try
+            {
+                ret = InvokeMember(memberName, flags, null, target, aArgs, aParamMod, null, null);
+            }
+            catch (TargetInvocationException e)
+            {
+                // For target invocation exceptions, we need to unwrap the inner exception and
```

Comment:
You can set `DoNotWrapExceptions` (new in CoreCLR) in the `flags` instead of doing this unwrapping.

Reply:
Yay!